plugin/trino-hive/src/test/java/io/trino/plugin/hive/metastore/glue/TestHiveGlueMetastore.java

```diff
@@ -207,6 +219,37 @@ protected HiveMetastore createMetastore(File tempDir)
                                 .setHideDeltaLakeTables(true)).get());
     }
 
+    @Test
+    public void cleanupOrphanedDatabases()
```

Comment:
Would make tests listing databases flaky. We should fix such tests. I didn't spend time to find such tests.

Aternatively make a method and call it from @BeforeClass. Still flaky across CI runs but lower chances.

Reply:
> Would make tests listing databases flaky.

This is a preexisting problem.
The cleanup of 'own' resources is supposed to happen in https://github.com/trinodb/trino/blob/4d0d1cf2a2f0baa64feffb71b1e16abf6b890f0c/plugin/trino-hive/src/test/java/io/trino/plugin/hive/AbstractTestHiveLocal.java#L117 , so we're already removing databases.

> Aternatively make a method and call it from @BeforeClass. Still flaky across CI runs but lower chances.

As you pointed out -- this would guard against concurrency within test class, but not with concurrent executions.
i would feel it's putting problem under the carpet, making it harder to notice, and so harder to fix.

>  I didn't spend time to find such tests.

I'd opt for merging as is.
If we find such tests, we probably find product issues. Eg `information_schema` should work even if databases (schemas) are dropped concurrently. Like https://github.com/trinodb/trino/pull/10844/ but with schemas going aways. Thus, finding a flaky test would be valuable.

Reply:
> https://github.com/trinodb/trino/blob/4d0d1cf2a2f0baa64feffb71b1e16abf6b890f0c/plugin/trino-hive/src/test/java/io/trino/plugin/hive/AbstractTestHiveLocal.java#L117

Ah, that makes sense.

Agreed with everything.

Reply:
> Agreed with everything.

so nice of you :)